manager/integration/tests/test_node.py

```diff
@@ -350,6 +350,22 @@ def test_replica_scheduler_no_disks(client):  # NOQA
     common.wait_for_volume_delete(client, vol_name)
 
 
+@pytest.mark.node  # NOQA
+def test_disable_scheduling_on_cordoned_node(client):  # NOQA
+    """
+    Test replica scheduler: schedule replica based on
+    `Disable Scheduling On Cordoned Node` setting
+
+    1. Set `Disable Scheduling On Cordoned Node` to true.
+    2. Set `Replica Soft Anti-Affinity` to false.
+    3. Set cordon on one node.
+    4. Create a volume and verify the scheduler should fail due to cordoned node.
```

Comment:
Why would we expect the scheduler to fail when one node is cordoned? It should be able to schedule the replicas on other nodes which are not cordoned.

Reply:
This is because we disable the soft anti-affinity, we shouldn't schedule multiple replicas on the same node.

Reply:
@boknowswiki But Scheduler will be able to schedule replicas on the node which is not cordoned isnt it?

Reply:
@sowmyav27 , the case here is we have 3 nodes, 1 is cordoned, and we need 3 replicas, and we have disabled the soft on node anti-affinity. So we can schedule 2 replicas on the 2 good nodes, but for the cordoned node, we should not schedule replica on it. Hence the scheduler failed to schedule 3 replicas. Please let me know what you think.

Thanks.